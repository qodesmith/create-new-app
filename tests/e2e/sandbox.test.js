const path = require('path')
const fs = require('fs-extra')
const run = require('../../modules/run')
const { listFolderContents } = require('./helpers/folderFileHelper')


describe('cli - sandbox project', () => {
  const appName = 'sandbox-test' // Ensure the folder ends in `test` to be git ignored.
  const mainPath = path.resolve(__dirname, '../../')
  const appPath = `${mainPath}/${appName}`
  const appContents = ['index.html', 'main.js', 'styles.css']

  beforeAll(() => {
    run(`node ${mainPath}/main.js ${appName} --sandbox`)
  })

  afterAll(() => {
    fs.removeSync(appPath)
  })

  it('should create a sandbox project in the <appName> directory', () => {
    expect(fs.existsSync(appPath)).toBe(true)
  })

  it('should only contain 3 files', () => {
    const folders = listFolderContents(appPath, { folders: true })
    const files = listFolderContents(appPath, { files: true, namesOnly: true })

    expect(folders).toEqual([])
    expect(files.sort()).toEqual(appContents.sort())
  })

  appContents.forEach(file => {
    it(`should populate "${file}" correctly`, () => {
      const fileContents = fs.readFileSync(`${appPath}/${file}`, 'utf8')
      expect(fileContents).toMatchSnapshot()
    })
  })
})
