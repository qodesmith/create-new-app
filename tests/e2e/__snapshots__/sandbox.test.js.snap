// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli - sandbox project should populate "index.html" correctly 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <title>Sandbox Project</title>
  <link rel=\\"stylesheet\\" href=\\"./styles.css\\">
</head>
<body>
  <script src=\\"./main.js\\"></script>
</body>
</html>
"
`;

exports[`cli - sandbox project should populate "main.js" correctly 1`] = `
"/*
  This is a great way to quickly test things out!

  Add all your JavaScript here.
  Open the \`index.html\` file in Chrome, save changes here,
  and simply refresh Chrome to see those changes.
*/

console.log('Loaded, bro.')
"
`;

exports[`cli - sandbox project should populate "styles.css" correctly 1`] = `
"/*
  Add all your css styling here.
*/
html,
body {
  padding: 0;
  margin: 0;
  font-family: sans-serif;
}
"
`;
